src/audio/codec_adapter/codec/waves.c

```diff
@@ -0,0 +1,681 @@
+		DUMP_REVISION_ITERATION(r32, l32);
+		DUMP_REVISION_ITERATION(r32, l32);
+		DUMP_REVISION_ITERATION(r32, l32);
+		DUMP_REVISION_ITERATION(r32, l32);
+	}
+
+	comp_info(dev, "waves_effect_revision() done");
+	return 0;
+}
+
+static int waves_effect_allocate(struct comp_dev *dev)
+{
+	// allocate additional buffers for MaxxEffect
+	struct codec_data *codec = comp_get_codec(dev);
+	struct waves_codec_data *waves_codec = codec->private;
+	MaxxStatus_t status;
+
+	comp_dbg(dev, "waves_effect_allocate() start");
+
+	// memmory for response
+	// SOF does not support get requests from codec adapter right now
+	// response will be stored in internal buffer to be dumped into logs
+	status = MaxxEffect_GetMessageMaxSize(waves_codec->effect,
+		&waves_codec->request_max_bytes, &waves_codec->response_max_bytes);
+
+	if (status) {
+		comp_err(dev, "waves_effect_allocate() MaxxEffect_GetMessageMaxSize() error %d",
+			status);
+		return -EIO;
+	}
+
+	waves_codec->response = codec_allocate_memory(dev, waves_codec->response_max_bytes, 16);
+	if (!waves_codec->response) {
+		comp_err(dev, "waves_effect_allocate() memory for response");
+		return -ENOMEM;
+	}
+
+	// memory for input/output buffers
+	waves_codec->i_buffer = codec_allocate_memory(dev, waves_codec->buffer_bytes, 16);
+	if (!waves_codec->i_buffer) {
+		codec_free_memory(dev, waves_codec->response);
+		comp_err(dev, "waves_effect_allocate() allocate memory for i_buffer");
+		return -ENOMEM;
+	}
+
+	waves_codec->o_buffer = codec_allocate_memory(dev, waves_codec->buffer_bytes, 16);
+	if (!waves_codec->o_buffer) {
+		codec_free_memory(dev, waves_codec->response);
+		codec_free_memory(dev, waves_codec->i_buffer);
+		comp_err(dev, "waves_effect_allocate() allocate memory for o_buffer");
+		return -ENOMEM;
+	}
+
+	comp_info(dev, "waves_effect_allocate() size response %d, i_buffer %d, o_buffer %d",
+		waves_codec->response_max_bytes,
+		waves_codec->buffer_bytes,
+		waves_codec->buffer_bytes);
+
+	codec->cpd.in_buff = waves_codec->i_buffer;
+	codec->cpd.in_buff_size = waves_codec->buffer_bytes;
+	codec->cpd.out_buff = waves_codec->o_buffer;
+	codec->cpd.out_buff_size = waves_codec->buffer_bytes;
+
+	comp_dbg(dev, "waves_effect_allocate() done");
+	return 0;
+}
+
+static int waves_effect_message(struct comp_dev *dev, void *data, uint32_t size)
+{
+	struct codec_data *codec = comp_get_codec(dev);
+	struct waves_codec_data *waves_codec = codec->private;
+	MaxxStatus_t status;
+	uint32_t response_size = 0;
+
+	comp_info(dev, "waves_effect_message() start data %p size %d", data, size);
+
+	status = MaxxEffect_Message(waves_codec->effect, data, size,
+		waves_codec->response, &response_size);
+
+	if (status) {
+		comp_err(dev, "waves_effect_message() MaxxEffect_Message() error %d",
+			status);
+		return -EIO;
+	}
+
+	// at time of writing codec adapter does not support getting something from codec
+	// so response is stored to internal structure and dumped into trace messages
+	if (response_size) {
```

Comment:
what is the purpose of this trace exactly? If it is strictly for debugging then we should probably put the loop behind a build flag

Reply:
our component usually provides a response after configuration is set.
since there is no way to deliver the response back to userspace we are dumping it into logs.

if this is a problem then it can be put behind a build flag

Reply:
Is the response actionable? As for the trace i am just worried about it not being optimized out in non-debug builds